src/DurableTask.AzureStorage/OrchestrationSessionManager.cs

```diff
@@ -174,6 +174,36 @@ async Task DequeueLoop(string partitionId, ControlQueue controlQueue, Cancellati
                 $"Stopped listening for messages on queue {controlQueue.Name}.");
         }
 
+        public async Task DrainAsync(string partitionId, CloseReason reason, CancellationToken cancellationToken)
+        {
+            // Start the drain process, mark the queue released to stop listening to new msg
```

Comment:
nit: for readability, I prefer not to shorten words in comments :)
```suggestion
            // Start the drain process, mark the queue released to stop listening for new messages
```